sdk/go/common/workspace/plugins_test.go

```diff
@@ -869,3 +905,19 @@ plugins:
 	assert.Equal(t, "1.0.0", proj.Plugins.Providers[0].Version)
 	assert.Equal(t, "../bin/aws", proj.Plugins.Providers[0].Path)
 }
+
+func TestPluginBadSource(t *testing.T) {
+	t.Parallel()
+
+	version := semver.MustParse("4.30.0")
+	spec := PluginSpec{
+		PluginDownloadURL: "strange-scheme://what.is.this?oh-no",
+		Name:              "mockdl",
+		Version:           &version,
+		Kind:              PluginKind("resource"),
+	}
+	source, err := spec.GetSource()
+	require.Error(t, err)
+	assert.Nil(t, source)
+	assert.Equal(t, "unknown plugin source scheme strange-scheme", err.Error())
```

Comment:
optional:

- can shorten the two error checks into a single ErrorContains or ErrorEquals -- they assert the error exists and then a condition about it
- asserting a nil source isn't necessary—the contract is that if err is non nil, source doesn't matter
With ErrorContains

```suggestion
	_, err := spec.GetSource()
	assert.ErrorContains(t, err, "unknown plugin source scheme strange-scheme")
```

With EqualError:

```suggestion
	_, err := spec.GetSource()
	assert.EqualEqual(t, err, "unknown plugin source scheme strange-scheme")
```

Reply:
> asserting a nil source isn't necessary—the contract is that if err is non nil, source doesn't matter

Well yes, that is the contract and the test is asserting that the method is sticking to that contract.

I'll change up the equals checks though, ErrorEquals saves a line.